plugins/dql-backend/src/service/router.ts

```diff
@@ -45,11 +49,19 @@ export const createRouter = async ({
   });
 
   router.get('/dynatrace/:queryId', async (req, res) => {
+    const { kubernetesId, labelSelector, componentNamespace, componentName } =
+      req.query;
+    validateQueryParameters(req.query, req.params.queryId);
     const deployments = await queryExecutor.executeDynatraceQuery(
       req.params.queryId,
       {
-        componentNamespace: req.query.componentNamespace as string,
-        componentName: req.query.componentName as string,
+        componentName: componentName as string,
+        componentNamespace: componentNamespace as string | undefined,
+        additionalFilter: generateComplexFilter(
```

Comment:
Right now, this is one big filter that allows filtering for K8s labels and namespaces.
I would split it.

Reply:
Split it into three